### PR TITLE
feat: domain trade and webhosting product

### PR DESCRIPTION
**Pull request by marcaudefroy**
feat: add push webhosting product into a cart
feat: add trade cli

### CODE REVIEW

File: ovh/ovhcli/order/domains.go [hunk 1]
```diff
@@ -67,3 +72,62 @@ var CmdDomain = &cobra.Command{
...
+				serviceOptionChoosed = &serviceOption
+				break
+			}
+		}
+		if serviceOptionChoosed == nil {
+			err = fmt.Errorf("Cannot find service options of type trade for domain %s", domain)
+			common.Check(err)
+		}
+		item, err := client.OrderAddCartServiceOption(domain, types.OrderCartServiceOptionDomainPost{
+			Duration:    "P1Y",
```
Comment (yesnault) [inline ovh/ovhcli/order/domains.go]: why not arg with default value ? or a comment

File: ovh/ovhcli/order/domains.go [hunk 2]
```diff
@@ -67,3 +72,62 @@ var CmdDomain = &cobra.Command{
...
+		common.Check(err)
+
+		err = client.OrderAssignCart(cart.CartID)
+		common.Check(err)
+
+		serviceOptions, err := client.OrderGetCartServiceOptions(domain)
+		common.Check(err)
+
+		var serviceOptionChoosed *types.OrderCartGenericOptionDefinition
+		for i := range serviceOptions {
```
Comment (yesnault) [inline ovh/ovhcli/order/domains.go]: for _, serviceOption := range servicesOptions { no ?

File: ovh/ovhcli/order/domains.go [hunk 3]
```diff
@@ -67,3 +72,62 @@ var CmdDomain = &cobra.Command{
...
+			serviceOption := serviceOptions[i]
+			if serviceOption.Family == "trade" {
+				serviceOptionChoosed = &serviceOption
+				break
+			}
+		}
+		if serviceOptionChoosed == nil {
+			err = fmt.Errorf("Cannot find service options of type trade for domain %s", domain)
```
Comment (yesnault) [inline ovh/ovhcli/order/domains.go]: common.Exit("Cannot find service options of type trade for domain %s", domain)

File: ovh/ovhcli/order/domains.go [hunk 4]
```diff
@@ -67,3 +72,62 @@ var CmdDomain = &cobra.Command{
...
+	Use:   "domainTrade <domain>",
+	Short: "Order domain trade",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			common.WrongUsage(cmd)
+		}
+		domain := args[0]
+
+		if withOwner == "" {
+			err := fmt.Errorf("Missing withOwner option")
```
Comment (yesnault) [inline ovh/ovhcli/order/domains.go]: common.Exit("Missing withOwner option") or common.WrongUsage by adding new arg for err description.

File: ovh/orderwebhosting.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"errors"
+
+	"github.com/runabove/go-sdk/ovh/types"
+)
+
+// OrderAddProductWebHosting post a new webHosting product in your cart
+func (c *Client) OrderAddProductWebHosting(cartID string, orderCartWebHostingPost types.OrderCartWebHostingPost) (*types.OrderCartItem, error) {
+	if cartID == "" {
+		return nil, errors.New("Error 404: \"Invalid Cart ID\"")
```
Comment (yesnault) [inline ovh/orderwebhosting.go]: why 404 and not just Cart ID cannot be null ?

**Review by yesnault — CHANGES_REQUESTED** [review]
Why 404 in err ?

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]
LGTM

**Pull request merged**